ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -13212,7 +13212,7 @@ private void updateDefaultTblProps(Map<String, String> source, Map<String, Strin
         retValue.put(hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES,
             TransactionalValidationListener.INSERTONLY_TRANSACTIONAL_PROPERTY);
       }
-      if (makeAcid || isTransactional) {
+      if (makeAcid || isTransactional || (isManaged && !makeInsertOnly)) {
```

Comment:
Do we have the case it is isTransaction = true, isManaged = true and makeInsertOnly is false (happen not to set TABLE_TRANSACTIONAL_PROPERTIES.) ?

Reply:
so isManaged and isTransactional cannot be true at the time. 
create managed table ... -->isManaged=true
create transactional table -->isTransactional=true

with either of the above set, makeInsertOnly can be true or false. This is the server side setting not driven by the query syntax.

so create managed creates "insert_only" acid table if "hive.create.as.insert.only"=true otherwise it creates a full acid table